### ISSUE TITLE
Upgrade to react-native 0.31.0

### ISSUE THREAD

**Issue by DanielZlotin** [wix/react-native-navigation#234]

**Comment by musicode** [comment]
is it complete?

**Comment by explorigin** [comment]
No, this is an issue to track the process.

**Comment by musicode** [comment]
how long to wait for it?

**Comment by DanielZlotin** [comment]
iOS is complete (RN31)
Working on Android

**Comment by musicode** [comment]
iOS RN32 available?

**Comment by DanielZlotin** [comment]
No, we have a weird bug when using anything higher than 31 on iOS. see https://github.com/wix/react-native-navigation/issues/257

**Comment by DanielZlotin** [comment]
update: RN31 works great on iOS, works OK on Android. 

There are indications of memory issues on Android but that's probably related to the temp hack we did with view unmounting/detaching that we need to remove. Easy peasy.

**Comment by bbrock25** [comment]
@DanielZlotin will you be releasing a version bump for this in the near future?

**Comment by DanielZlotin** [comment]
@bbrock25 iOS is already out (with the controllers update), Android should be soon. Unfortunately, we are blocked by the fact we use this framework along with lots of other libraries in production, some of which do not support RN>25. we are working hard on migrating.

**Comment by DanielZlotin** [comment]
By soon, I mean we plan doing it next Monday. No guarantees yet.

**Comment by musicode** [comment]
> By soon, I mean we plan doing it next Monday. No guarantees yet.

how about RN32?

**Comment by explorigin** [comment]
Hey guys, I just wanted to say Thanks!  This is a much needed package in the RN space.  Please pass on my gratitude to your respective employers for offering your time in such a way that we can all benefit from.

**Comment by DanielZlotin** [comment]
@musicode as said in https://github.com/wix/react-native-navigation/issues/257#issuecomment-242979908 - I started seeing weird issues of views not appearing when going from 31 to 32 on iOS. So as I see it now, we will upgrade to RN31 first in order not to jump too high all at once.

**Comment by DanielZlotin** [comment]
updated to RN31 in `v2.0.0-experimental.91`.
We need to remove some left over hacks, I'll open a separate issue. Please report any resulting difficulties.

**Comment by DanielZlotin** [comment]
Breaking changes:

Nothing in native (we hide the implementation for you in our NavigationApplication), but when moving your project to RN31, you should know about a few RN breaking changes:

* importing `React, {Component, PropTypes}` must be from `react` and not `react-native` (used to throw warning, will blow up now)
* `ActivityIndicatorIOS` is deprecated, replace with `ActivityIndicator` (both platforms are supported now)
* specifying `flex: 1` in places that you don't need it will now break your ui completely
* specifying `FlexWrap: wrap` without `alignItems: flex-start` will now break you ui

**Comment by AdamBrodzinski** [comment]
@DanielZlotin Great to hear! and keep up the great work!! If there's anything I can test specifically let me know, going to try this out tomorrow. 👍
